### PR TITLE
Us16 s2 make artwork page

### PR DESCRIPTION
**Pull request by R-Ztr**

### CODE REVIEW

**Comment by StoicFrenchDev** [comment]
Label!

**Review by Guadanino — APPROVED** [review]

File: templates/artwork/index.html.twig [hunk 1]
```diff
@@ -1,47 +1,25 @@
...
-        {% else %}
-            <tr>
-                <td colspan="9">no records found</td>
-            </tr>
-        {% endfor %}
-        </tbody>
-    </table>
+<h2>Oeuvres</h2>
 
+<div class="gallery-container">
```
Comment (StoicFrenchDev) [inline templates/artwork/index.html.twig:9]: Unclosed tag

File: templates/artwork/index.html.twig [hunk 2]
```diff
@@ -1,47 +1,25 @@
...
+    {% for artwork in artworks %}
+        <div class="gallery">
+            <img class=" image-artwork" alt="{{ artwork.description }}" src="{{ artwork.imageCover }}"> 
+            <p>{{ artwork.title }}</br>
+            {{ artwork.description }}</p>
+            <button class="btn btn-secondary form-button"><a class="form-link" href="{{ path('app_artwork_show', {'id': artwork.id}) }} ">Découvrir</a></button>
+            {#<button class="btn btn-secondary form-button"><a class="form-link" href="{{ path('app_artwork_edit', {'id': artwork.id}) }}">edit</a></button>#}
```
Comment (StoicFrenchDev) [inline templates/artwork/index.html.twig:16]: Dead code

File: templates/artwork/index.html.twig [hunk 3]
```diff
@@ -1,47 +1,25 @@
...
-    </table>
+<h2>Oeuvres</h2>
 
+<div class="gallery-container">
+    {% for artwork in artworks %}
+        <div class="gallery">
+            <img class=" image-artwork" alt="{{ artwork.description }}" src="{{ artwork.imageCover }}"> 
+            <p>{{ artwork.title }}</br>
+            {{ artwork.description }}</p>
+            <button class="btn btn-secondary form-button"><a class="form-link" href="{{ path('app_artwork_show', {'id': artwork.id}) }} ">Découvrir</a></button>
```
Comment (StoicFrenchDev) [inline templates/artwork/index.html.twig:15]: You must not put an "a" tag within a button tag

File: templates/artwork/index.html.twig [hunk 4]
```diff
@@ -1,47 +1,25 @@
...
-        {% endfor %}
-        </tbody>
-    </table>
+<h2>Oeuvres</h2>
 
+<div class="gallery-container">
+    {% for artwork in artworks %}
+        <div class="gallery">
+            <img class=" image-artwork" alt="{{ artwork.description }}" src="{{ artwork.imageCover }}"> 
+            <p>{{ artwork.title }}</br>
```
Comment (StoicFrenchDev) [inline templates/artwork/index.html.twig:13]: Why a "br" tag here?

File: templates/artwork/indexByCategory.html.twig [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{% extends 'base.html.twig' %}
+
+    {% block body %}
+
+        <h2>Oeuvres : {{ category.name }}</h2>
+
+        <div class="gallery-container">
```
Comment (StoicFrenchDev) [inline templates/artwork/indexByCategory.html.twig:7]: Unclosed tag

File: templates/artwork/indexByCategory.html.twig [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        {% for artwork in artworks %}
+            <div class="gallery">
+                <img class=" image-artwork" alt="{{ artwork.description }}" src="{{ artwork.imageCover }}"> 
+                <p>{{ artwork.title }}</br>
+                {{ artwork.description }}</p>
+                <button class="btn btn-secondary form-button"><a class="form-link" href="{{ path('app_artwork_show', {'id': artwork.id}) }} ">Découvrir</a></button>
```
Comment (StoicFrenchDev) [inline templates/artwork/indexByCategory.html.twig:13]: You must not put an "a" tag within a button tag

File: templates/artwork/index.html.twig [hunk 7]
```diff
@@ -1,47 +1,25 @@
...
+            {{ artwork.description }}</p>
+            <button class="btn btn-secondary form-button"><a class="form-link" href="{{ path('app_artwork_show', {'id': artwork.id}) }} ">Découvrir</a></button>
+            {#<button class="btn btn-secondary form-button"><a class="form-link" href="{{ path('app_artwork_edit', {'id': artwork.id}) }}">edit</a></button>#}
+        </div>
+    {% else %}
+        <tr>
```
Comment (StoicFrenchDev) [inline templates/artwork/index.html.twig:19]: Why "tr" and "td" tags?

File: templates/artwork/indexByCategory.html.twig [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+            </div>      
+        {% else %}
+            <tr>
```
Comment (StoicFrenchDev) [inline templates/artwork/indexByCategory.html.twig:16]: Why "tr" and "td" tags?

File: templates/home/index.html.twig [hunk 9]
```diff
@@ -39,39 +39,39 @@ créatifs du monde entier."</p>
 					<img class=" image" alt="Photo d'un panorama" src="{{ asset('build/images/janez.webp') }}"  >
 					<div class="overlay">
 						<h3>Photos</h3>
-            <a href="#">Découvrir</a>
+            <a href="artwork/category/photo">Découvrir</a>
             <p></p>
```
Comment (StoicFrenchDev) [inline templates/home/index.html.twig:43]: Empty "p" tag

File: templates/home/index.html.twig [hunk 10]
```diff
@@ -39,39 +39,39 @@ créatifs du monde entier."</p>
...
 					</div>
 				</div>
 				<div class="box">
 					<img class= " image" alt="Peinture d'une femme" src="{{ asset('build/images/caravage.webp') }}">
 					<div class="overlay">
 						<h3>Peintures</h3>
-						<a href="#">Découvrir</a>
+						<a href="artwork/category/peinture">Découvrir</a>
             <p></p>
```
Comment (StoicFrenchDev) [inline templates/home/index.html.twig:51]: Empty "p" tag

File: templates/home/index.html.twig [hunk 11]
```diff
@@ -39,39 +39,39 @@ créatifs du monde entier."</p>
...
 					</div>
 				</div>
 				<div class="box">
 					<img class=" image" alt="Sculture d'écureuil" src="{{ asset('build/images/squirrel.png') }}">
 					<div class="overlay">
 						<h3>Sculptures</h3>
-						<a href="#">Découvrir</a>
+						<a href="artwork/category/sculpture">Découvrir</a>
             <p></p>
```
Comment (StoicFrenchDev) [inline templates/home/index.html.twig:59]: Empty "p" tag

File: templates/home/index.html.twig [hunk 12]
```diff
@@ -39,39 +39,39 @@ créatifs du monde entier."</p>
...
 					</div>
 				</div>
         <div class="box">
 					<img class=" image" alt="Fan art de Luffy" src="{{ asset('build/images/luffy.webp') }}">
 					<div class="overlay">
 						<h3>Fan Arts</h3>
-						<a href="#">Découvrir</a>
+						<a href="artwork/category/fanart">Découvrir</a>
             <p></p>
```
Comment (StoicFrenchDev) [inline templates/home/index.html.twig:67]: Empty "p" tag

File: templates/home/index.html.twig [hunk 13]
```diff
@@ -39,39 +39,39 @@ créatifs du monde entier."</p>
...
 					</div>
 				</div>
         <div class="box">
 					<img class=" image" alt="Image générée par IA" src="{{ asset('build/images/colorBou.webp') }}">
 					<div class="overlay">
-						<h3>Numériques</h3>
-						<a href="#">Découvrir</a>
+						<h3>IA</h3>
+						<a href="artwork/category/ia">Découvrir</a>
             <p></p>
```
Comment (StoicFrenchDev) [inline templates/home/index.html.twig:75]: Empty "p" tag

**Review by StoicFrenchDev — CHANGES_REQUESTED** [review]

File: templates/artwork/indexByCategory.html.twig [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+
+    <h2>Oeuvres : {{ category.name }}</h2>
+
+    <div class="gallery-container">
+    {% for artwork in artworks %}
+        <div class="gallery">
+            <img class=" image-artwork" alt="{{ artwork.description }}" src="{{ artwork.imageCover }}"> 
+                <p>{{ artwork.title }}</p>
+                    <p>{{ artwork.description }}</p>
+                    <button class="btn btn-secondary form-button">
```
Comment (StoicFrenchDev) [inline templates/artwork/indexByCategory.html.twig:13]: Still not accepting an "a" tag within a button tag

File: templates/artwork/index.html.twig [hunk 15]
```diff
@@ -1,47 +1,24 @@
...
-    </table>
+<h2>Oeuvres</h2>
 
+<div class="gallery-container">
+    {% for artwork in artworks %}
+        <div class="gallery">
+            <img class=" image-artwork" alt="{{ artwork.description }}" src="{{ artwork.imageCover }}"> 
+                <p>{{ artwork.title }}</p>
+                    <p>{{ artwork.description }}</p>
+                        <button class="btn btn-secondary form-button">
```
Comment (StoicFrenchDev) [inline templates/artwork/index.html.twig:15]: Still not accepting an "a" tag within a button tag

**Review by StoicFrenchDev — CHANGES_REQUESTED** [review]

**Comment by R-Ztr** [comment]
valide stp c'est pas mon code le button

Le jeu. 21 déc. 2023 à 16:41, Benjamin RICHARD ***@***.***> a
écrit :

**Review by StoicFrenchDev — APPROVED** [review]

**Pull request merged**